### PR TITLE
remove duplication in tomcat JMX config

### PR DESCRIPTION
**Pull request by SylvainJuge**
The YAML configuration allows to use multiple MBeans definitions at once, which allows to avoid duplication when querying MBeans that have the same "shape" and for which the MBeans attributes to query are the same.

Here is an [existing example for Kafka](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/13e148b847a7a0fc7df7f95198a48fe2cfe87c3d/instrumentation/jmx-metrics/javaagent/src/main/resources/jmx/rules/kafka-broker.yaml#L53).

This allows to elegantly remove the duplication introduced in #10115 for Tomcat that can be exposed with either `Catalina:` or `Tomcat:` as JMX domain.

### CODE REVIEW

**Review by laurit — APPROVED** [review]

**Review by heyams — APPROVED** [review]

**Review by robsunday — APPROVED** [review]
All is good

**Comment by trask** [comment]
cc @PeterF778 in case you have a chance to review

File: instrumentation/jmx-metrics/javaagent/src/main/resources/jmx/rules/tomcat.yaml [hunk 1]
```diff
@@ -4,45 +4,9 @@
 # would match too broadly
```
Comment (PeterF778) [inline instrumentation/jmx-metrics/javaagent/src/main/resources/jmx/rules/tomcat.yaml:4]: Replace "thus we have to duplicate metrics definitions as using a wildcard would match too broadly" by "thus we use both MBean names for most of the metrics"

**Comment by PeterF778** [comment]
Looks good to me.

**Review by PeterF778 — APPROVED** [review]

**Pull request merged**